cmake/modules/SearchInstalledSoftware.cmake

```diff
@@ -1507,12 +1507,12 @@ endif()
 if(tmva AND cuda AND tmva-gpu)
   message(STATUS "Looking for CUDA for optional parts of TMVA")
 
-  if(cxx11)
+  if(CMAKE_CXX_STANDARD EQUAL 11)
     find_package(CUDA 7.5)
-  elseif(cxx14)
+  elseif(CMAKE_CXX_STANDARD EQUAL 14)
     message(STATUS "Detected request for c++14, requiring minimum version CUDA 9.0 (default 7.5)")
     find_package(CUDA 9.0)
-  elseif(cxx17)
+  elseif(CMAKE_CXX_STANDARD EQUAL 17)
```

Comment:
What will happen for 20? Why not just `else()`?

Reply:
We need support in other places like `root-config`, `R__HAS_CXX20`, etc, before we can let people use C++20, that's why I added the message. We can fix that before 6.18 is out, though, but it will require more work. We also need to make sure nothing that has been removed is still used by ROOT, and so on.

Reply:
Ideally, ROOT (not caring about > 17 but only about >= 17) should set `R__HAS_CXX17` when compiled with C++20. That shows a design mistake in these `usec++...`: we should always define the earlier ones, too, such that for `CMAKE_CXX_STANDARD` >= 17, all current ones are defined.

This should probably go into a separate PR...

Reply:
We agreed to rip out `R__USE_CXX` from ROOT; it was ill-defined (e.g. `R__USE_CXX14` was not defined for C++17) and people should not rely on it, but use `__cplusplus` instead. Which is what we'll do in the single use in ROOT. @amadio offered a subsequent PR.

Reply:
Actually, it was easy enough that I just added this to the current PR (removal of `R__USE_CXX*`).